Production/python/Production.py

```diff
@@ -251,6 +251,9 @@
         getattr(process, producerName).badPFMuonFilter = cms.InputTag('BadPFMuonFilter')
         getattr(process, producerName).badChCandidateFilter = cms.InputTag('BadChargedCandidateFilter')
 
+    if options.runKinFit:
+        getattr(process, producerName).rho = cms.InputTag('fixedGridRhoAll')
```

Comment:
rho is used in JER calculation, which is stored in EventTuple independently on options.runKinFit. Fix it here and in the c++ code.